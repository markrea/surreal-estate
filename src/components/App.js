import React from "react";
import { Route, Switch } from "react-router-dom";
import "../styles/App.css";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";

function App() {
  return (
    <div>
      <NavBar />
      <Switch>
        <Route exact path="/" component={Properties} />
        <Route exact path="/add-property" component={AddProperty} />
      </Switch>
    </div>
  );
}

export default App;
