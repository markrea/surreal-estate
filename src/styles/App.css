@import url("https://fonts.googleapis.com/css2?family=Poppins&display=swap");

* {
  box-sizing: border-box;
}

.app {
display: grid;
grid-template-columns: 1fr 3fr;
grid-template-rows: 1fr 5fr;

  
}
body {
  font-family: "Poppins", sans-serif;
  margin: 0;
  padding: 0;
  background-color: lightgrey;
  
  margin: auto;
}
 
 
