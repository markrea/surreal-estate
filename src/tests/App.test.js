import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import App from "../components/App";

test("component renders correctly", () => {
  const history = createMemoryHistory();
  const { asFragment } = render(
    <Router history={history}>
      <App />
    </Router>
  );
  const component = asFragment();
  expect(component).toMatchSnapshot();
});
