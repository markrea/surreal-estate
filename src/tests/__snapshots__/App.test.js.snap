// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="navbar"
      data-testid="nav-bar-id"
    >
      <img
        alt="SELogo"
        class="SE-logo"
        data-testid="logo-id"
        src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
      />
      <ul
        class="navbar-links"
      >
        <li>
          <a
            class="navbar-links-item"
            href="/"
          >
            View Properties
          </a>
        </li>
        <li>
          <a
            class="navbar-links-item"
            href="/add-property"
          >
            Add a Property
          </a>
        </li>
      </ul>
    </div>
    <div
      class="property-body"
    >
      <div>
        <a
          class="sidebar-link"
          href="/?query={\\"city\\":\\"Manchester\\"}"
        >
          Manchester
        </a>
        <a
          class="sidebar-link"
          href="/?query={\\"city\\":\\"Liverpool\\"}"
        >
          Liverpool
        </a>
        <a
          class="sidebar-link"
          href="/?query={\\"city\\":\\"Sheffield\\"}"
        >
          Sheffield
        </a>
        <a
          class="sidebar-link"
          href="/?query={\\"city\\":\\"Leeds\\"}"
        >
          Leeds
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
